src/transformers/models/sam/modeling_sam.py

```diff
@@ -865,7 +881,8 @@ def window_partition(self, hidden_states: torch.Tensor, window_size: int) -> Tup
 
         pad_h = (window_size - height % window_size) % window_size
         pad_w = (window_size - width % window_size) % window_size
-        if pad_h > 0 or pad_w > 0:
+        # TODO: the if statement can be removed.
+        if True: # pad_h > 0 or pad_w > 0:
```

Comment:
Same question as for interpolate

Reply:
Yes we can remove it as `F.pad` will do nothing if pad_h and pad_w == 0